cpg-analysis/src/main/kotlin/de/fraunhofer/aisec/cpg/query/FlowQueries.kt

```diff
@@ -494,7 +532,16 @@ fun Node.alwaysFlowsTo(
                         terminationReason = Success(it.nodes.last()),
                     )
                 }
-        nothingFailed = nothingFailed && nextEOGEvaluation.failed.isEmpty()
+        nothingFailed =
+            nothingFailed &&
+                nextEOGEvaluation.failed.none {
```

Comment:
can we extract the negation making it .all (noSinkIsGood && it.first == FailureReason.PATH_ENDED). That would IMO make it a lot more readable and AFAIK it should evaluate to the same expression